### PR TITLE
Added Pydantic validation function and manual validation documentation

### PR DESCRIPTION
**Pull request by kilianbartz**
I also fixed a small bug in folder loader

### CODE REVIEW

File: cbrkit/loaders.py [hunk 1]
```diff
@@ -341,23 +364,54 @@ def folder(path: Path, pattern: str) -> Casebase[Any, Any] | None:
...
+        >>> data = path(Path("data/cars-1k.csv"))
+        >>> validate(data, Car)
+        >>> import pandas as pd
+        >>> df = pd.read_csv("data/cars-1k.csv")
+        >>> data = dataframe(df)
+        >>> validate(data, Car)
+    """
+    if data is None:
+        raise ValueError("Data is None")
+    if isinstance(data, DataFrameCasebase):
```
Comment (mirkolenz) [inline cbrkit/loaders.py:411]:
```suggestion
    elif isinstance(data, DataFrameCasebase):
```

File: cbrkit/loaders.py [hunk 2]
```diff
@@ -341,23 +364,54 @@ def folder(path: Path, pattern: str) -> Casebase[Any, Any] | None:
...
+        data = data.df.to_dict("index")
```
Comment (mirkolenz) [inline cbrkit/loaders.py:412]: This is extremely slow. Is there a way to just iterate over all Series entries instead?

File: cbrkit/loaders.py [hunk 3]
```diff
@@ -341,23 +364,54 @@ def folder(path: Path, pattern: str) -> Casebase[Any, Any] | None:
...
+    if isinstance(data, dict):
```
Comment (mirkolenz) [inline cbrkit/loaders.py:413]:
```suggestion
    if isinstance(data, Mapping):
```

File: cbrkit/loaders.py [hunk 4]
```diff
@@ -1,3 +1,18 @@
+"""
+To manually use Pydantic with CBRkit to validate your case base, you can use an appropriate 
+Pydantic model instead of the CBRkit loaders (see example below). 
+Alternatively, the dataframe, path, file and folder accept an optional validation_model argument
+to validate the Casebase entries.
```
Comment (mirkolenz) [inline cbrkit/loaders.py:5]: The description should be updated for the new `cbrkit.loaders.validate` function.

File: cbrkit/loaders.py [hunk 5]
```diff
@@ -325,6 +342,12 @@ def file(path: Path) -> Casebase[Any, Any] | None:
         >>> from pathlib import Path
         >>> file_path = Path("./data/cars-1k.csv")
         >>> result = file(file_path)
+
+        >>> from pydantic import BaseModel, PositiveInt, NonNegativeInt
+        >>> from pathlib import Path
+        >>> file_path = Path("./data/cars-1k.csv")
+        >>> result = file(file_path)
+
```
Comment (mirkolenz) [inline cbrkit/loaders.py:350]: What does this snippet achieve? The model does not seem to be used here.

File: cbrkit/loaders.py [hunk 6]
```diff
@@ -341,23 +364,54 @@ def folder(path: Path, pattern: str) -> Casebase[Any, Any] | None:
...
     Returns:
         Returns a Casebase.
 
     Examples:
         >>> from pathlib import Path
+        >>> from data.cars_validation_model import Car
         >>> folder_path = Path("./data")
-        >>> result = folder(folder_path, ".csv")
+        >>> result = folder(folder_path, "*.csv")
+        >>> assert result is not None
```
Comment (mirkolenz) [inline cbrkit/loaders.py:376]: Same as above, what is the goal here?

**Review by mirkolenz — CHANGES_REQUESTED** [review]
Thanks for the PR, there are some things that should be changed before merging it. Great that you spotted the bug in the folder loader method!

File: cbrkit/loaders.py (hunk 2 above)
Comment (kilianbartz) [reply cbrkit/loaders.py:412]: It is possible to do it in the following manner:
`for item in df.iterrows():
    validation_model(**item[1])`
However, this is much slower at 50ms vs 6ms using df.to_dict().

Comment (mirkolenz) [reply cbrkit/loaders.py:395]: Then just leave it as it is for the time being. In case someone reports performance issues we can revisit it in the future.

**Review by mirkolenz — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**